pkg/mod/mod.go

```diff
@@ -0,0 +1,40 @@
+package mod
+
+import (
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+type goModule struct {
+	Path     string     // module path
+	Version  string     // module version
+	Versions []string   // available module versions (with -versions)
+	Replace  *goModule  // replaced by this module
+	Time     *time.Time // time version was created
+	Update   *goModule  // available update, if any (with -u)
+	Main     bool       // is this the main module?
+	Indirect bool       // is this module only an indirect dependency of main module?
+	Dir      string     // directory holding files for this module, if any
+	GoMod    string     // path to go.mod file for this module, if any
```

Comment:
These comments are not really available when someone is reading code later on that uses this struct, can you think of better field names so these comments are not needed.

i.e, lets say we have a variable of this type called `mod`, `mod.Path` is a fairly obvious name, `mod.Versions` and `mod.Version` is confusing (why have 2?).

`mod.Replace`? does this mean the mod is replaced by the one pointed to, or is this one replacing the pointed to one?

why `Path` and `Dir`?

Getting an API right is hard and extremely important, practice (and having to use bad API) will make this easy eventually.


Additionally, almost all of these don't appear to be used, YAGNI (look it up if you havnt seen that acronym before)

Reply:
Thanks for the detailed feedback! I'll rethink it.